src/builder/create_scheduled_event.rs

```diff
@@ -99,18 +118,50 @@ impl CreateScheduledEvent {
     /// on a path that doesn't exist.
     #[cfg(feature = "model")]
     pub async fn image<'a>(
-        &mut self,
+        mut self,
         http: impl AsRef<Http>,
         image: impl Into<AttachmentType<'a>>,
-    ) -> Result<&mut Self> {
+    ) -> Result<Self> {
         let image_data = image.into().data(&http.as_ref().client).await?;
-
-        self.image = Some(format!("data:image/png;base64,{}", base64::encode(image_data)));
+        self.fields.image = Some(format!("data:image/png;base64,{}", base64::encode(image_data)));
         Ok(self)
     }
+
+    /// Creates a new scheduled event in the guild with the data set, if any.
+    ///
+    /// **Note**: Requres the [Manage Events] permission.
+    ///
+    /// # Errors
+    ///
+    /// If the `cache` is enabled, returns a [`ModelError::InvalidPermissions`] if the current user
+    /// does not have permission to manage scheduled events.
+    ///
+    /// Otherwise will return [`Error::Http`] if the current user does not have permission.
+    ///
+    /// [Manage Events]: Permissions::MANAGE_EVENTS
+    pub async fn execute(self, cache_http: impl CacheHttp) -> Result<ScheduledEvent> {
+        #[cfg(feature = "cache")]
+        {
+            if let Some(cache) = cache_http.cache() {
+                if let Some(guild) = cache.guild(self.id) {
+                    let req = Permissions::MANAGE_EVENTS;
+
+                    if !guild.has_perms(&cache_http, req).await {
+                        return Err(Error::Model(ModelError::InvalidPermissions(req)));
+                    }
+                }
+            }
+        }
+
+        self._execute(cache_http.http()).await
+    }
+
+    async fn _execute(self, http: impl AsRef<Http>) -> Result<ScheduledEvent> {
```

Comment:
True, although its a private method, and will only ever be given an `Http`. I could remove the `AsRef` and it would still work. I'm planning on eventually factoring all the cache lookups into a function and replacing them all with a one-liner like `utils::check_perms(&cache_http, Permissions::MANAGE_EVENTS).await?`.